torch/csrc/jit/codegen/cuda/fusion.cpp

```diff
@@ -261,34 +261,43 @@ void Fusion::addInput(Val* const input) {
 
   if (input->getValType().value() == ValType::TensorView) {
     auto tv = input->as<TensorView>();
-    if (tv->hasReduction())
+    if (tv->hasReduction()) {
       TORCH_WARN_ONCE(
           "Registered input ",
           input,
           " has a reduction axis, but this does nothing in the fusion.");
+    }
+    tv->setMemoryType(MemoryType::Global);
+    TORCH_INTERNAL_ASSERT(
```

Comment:
Can this assert be removed as you just set it?

Reply:
done.